coalib/bearlib/abstractions/Linter.py

```diff
@@ -57,6 +57,33 @@ def _prepare_options(options):
 
         options["output_regex"] = re.compile(options["output_regex"])
 
+        supported_names = {
+            'origin',
+            'message',
+            'severity',
+            'line',
+            'column',
+            'end_line',
+            'end_column',
+            'additional_info'
+        }
+
+        no_of_non_named_groups = options["output_regex"].groups
+            - len(options["output_regex"].groupindex)
+        if no_of_non_named_groups:
+            logging.warn("Using unnecessary capture groups"
+                         " affects the performance of coala. "
+                         "You should use ``(?:<pattern>)`` instead of"
+                         " ``(<pattern>)`` for your regex.")
+        else:
+            for capture_group_name in options["output_regex"].groupindex:
+                if capture_group_name not in supported_names:
+                    logging.warn("Using unnecessary capture groups"
```

Comment:
```
Unknown named capturing group 'xxx' used. 'xxx-bear' will ignore it.
```
You assume that the name of capture group and the bear name is same :thinking: ?